test/unit/alphabet/structure_test.cpp

```diff
@@ -261,13 +270,15 @@ TEST(rna_structure, wuss)
         EXPECT_FALSE(vec[idx].is_unpaired());
         EXPECT_FALSE(vec[idx].is_pair_close());
         EXPECT_TRUE(vec[idx].pseudoknot_id());
+        EXPECT_EQ(*vec[idx].pseudoknot_id(), (idx - 7) / 2);
     }
     for (unsigned idx = 8; idx <= 22; idx+=2)
     {
         EXPECT_TRUE(vec[idx].is_pair_close());
         EXPECT_FALSE(vec[idx].is_unpaired());
         EXPECT_FALSE(vec[idx].is_pair_open());
         EXPECT_TRUE(vec[idx].pseudoknot_id());
+        EXPECT_EQ(*vec[idx].pseudoknot_id(), (idx - 8) / 2);
```

Comment:
Why fo you need to dereference it? Is the id a pointer?
// Ah okay it is an optional

Can you make the right side the return type of `pseudoknot_id()`?

So change it to:
```c++
EXPECT_EQ(vec[idx].pseudoknot_id(), std::optional{(idx - 8) / 2});
```

Reply:
Okay, I agree to make comparisons of same types.